cli/internal/lockfile/pnpm_lockfile.go

```diff
@@ -14,82 +14,94 @@ import (
+	Version                   interface{}                `yaml:"lockfileVersion"`
 	NeverBuiltDependencies    []string                   `yaml:"neverBuiltDependencies,omitempty"`
 	OnlyBuiltDependencies     []string                   `yaml:"onlyBuiltDependencies,omitempty"`
 	Overrides                 map[string]string          `yaml:"overrides,omitempty"`
 	PackageExtensionsChecksum string                     `yaml:"packageExtensionsChecksum,omitempty"`
 	PatchedDependencies       map[string]PatchFile       `yaml:"patchedDependencies,omitempty"`
+	Importers                 map[string]ProjectSnapshot `yaml:"importers"`
 	Packages                  map[string]PackageSnapshot `yaml:"packages,omitempty"`
 	Time                      map[string]string          `yaml:"time,omitempty"`
 }
 
 var _ Lockfile = (*PnpmLockfile)(nil)
-var _ Lockfile = (*PnpmLockfileV6)(nil)
 
 // ProjectSnapshot Snapshot used to represent projects in the importers section
 type ProjectSnapshot struct {
-	Specifiers           map[string]string           `yaml:"specifiers"`
-	Dependencies         map[string]string           `yaml:"dependencies,omitempty"`
-	OptionalDependencies map[string]string           `yaml:"optionalDependencies,omitempty"`
-	DevDependencies      map[string]string           `yaml:"devDependencies,omitempty"`
-	DependenciesMeta     map[string]DependenciesMeta `yaml:"dependenciesMeta,omitempty"`
-	PublishDirectory     string                      `yaml:"publishDirectory,omitempty"`
+	// for v6 we omitempty
+	// for pre v6 we *need* to omit the empty map
+	Specifiers SpecifierMap `yaml:"specifiers,omitempty"`
+
+	// The values of these maps will be string if lockfileVersion <6 or DependencyV6 if 6+
+	Dependencies         map[string]yaml.Node `yaml:"dependencies,omitempty"`
+	OptionalDependencies map[string]yaml.Node `yaml:"optionalDependencies,omitempty"`
+	DevDependencies      map[string]yaml.Node `yaml:"devDependencies,omitempty"`
+
+	DependenciesMeta map[string]DependenciesMeta `yaml:"dependenciesMeta,omitempty"`
+	PublishDirectory string                      `yaml:"publishDirectory,omitempty"`
 }
 
-// ProjectSnapshotV6 Snapshot used to represent projects in the importers section
-type ProjectSnapshotV6 struct {
-	Specifiers           map[string]DependencyV6     `yaml:"specifiers,omitempty"`
-	Dependencies         map[string]DependencyV6     `yaml:"dependencies,omitempty"`
-	OptionalDependencies map[string]DependencyV6     `yaml:"optionalDependencies,omitempty"`
-	DevDependencies      map[string]DependencyV6     `yaml:"devDependencies,omitempty"`
-	DependenciesMeta     map[string]DependenciesMeta `yaml:"dependenciesMeta,omitempty"`
-	PublishDirectory     string                      `yaml:"publishDirectory,omitempty"`
+// SpecifierMap is a type wrapper that overrides IsZero for Golang's map
+// to match the behavior that pnpm expects
+type SpecifierMap map[string]string
+
+// IsZero is used to check whether an object is zero to
```

Comment:
Out of curiosity, what is the default functionality of `IsZero`? If a map is nil or empty?

Reply:
The default behavior is `len(map) == 0` which is true if the map is nil or empty, but we need to differentiate between the two.